xen/arch/x86/smpboot.c

```diff
@@ -322,41 +315,24 @@ static void set_cpu_sibling_map(unsigned int cpu)
     }
 }
 
-void start_secondary(void *unused)
+void start_secondary(unsigned int cpu)
 {
     struct cpu_info *info = get_cpu_info();
 
+    /* Tell BSP that we are awake. */
+    set_cpu_state(cpu, CPU_STATE_INIT);
+
     /*
-     * Dont put anything before smp_callin(), SMP booting is so fragile that we
+     * Don't put anything before smp_callin(), SMP booting is so fragile that we
      * want to limit the things done here to the most necessary things.
      */
-    unsigned int cpu = booting_cpu;
-
-    if ( sl_status ) {
-        uint64_t misc_enable;
-        uint32_t my_apicid;
-        struct txt_sinit_mle_data *sinit_mle =
-              txt_sinit_mle_data_start(__va(read_txt_reg(TXTCR_HEAP_BASE)));
-
-        /* TXT released us with MONITOR disabled in IA32_MISC_ENABLE. */
-        rdmsrl(MSR_IA32_MISC_ENABLE, misc_enable);
-        wrmsrl(MSR_IA32_MISC_ENABLE,
-               misc_enable | MSR_IA32_MISC_ENABLE_MONITOR_ENABLE);
```

Comment:
No longer needed? Why?

Reply:
`MSR_IA32_MISC_ENABLE` for APs is written with value read from BSP in one of further initialization steps. This had to be done earlier to stop APs from proceeding, but it's no longer the case.